src/main/java/com/jcabi/maven/plugin/AjcMojo.java

```diff
@@ -187,7 +200,35 @@ public void execute() throws MojoFailureException {
         if (this.classesDirectory.mkdirs()) {
             Logger.info(this, "Created classes dir %s", this.classesDirectory);
         }
-        if (this.hasClasses() || this.hasSourceroots()) {
+        final boolean projectHasClasses = this.hasClasses();
```

Comment:
@amihaiemil can we also extract logic below to a method or create 15min puzzle for that? It's getting bigger and bigger